repo/maintenance/maintenance_quick_test.go

```diff
@@ -0,0 +1,217 @@
+
+	maintParams.Owner = co.UsernameAtHost()
+	maintenance.SetParams(ctx, env.RepositoryWriter, maintParams)
+
+	require.NoError(t, err)
+	require.NoError(t, env.RepositoryWriter.Flush(ctx))
+
+	maintParams, err = maintenance.GetParams(ctx, env.Repository)
+	require.NoError(t, err)
+	require.Equal(t, co.UsernameAtHost(), maintParams.Owner)
+
+	// verify epoch manager is enabled
+	dr, isDirect := env.Repository.(repo.DirectRepository)
+	require.True(t, isDirect)
+	require.NotNil(t, dr)
+
+	fm := dr.FormatManager()
+	require.NotNil(t, fm)
+
+	mp, err := fm.GetMutableParameters(ctx)
+	require.NoError(t, err)
+	require.True(t, mp.EpochParameters.Enabled, "epoch manager not enabled")
+
+	emgr, enabled, err := dr.ContentReader().EpochManager(ctx)
+	require.NoError(t, err)
+	require.True(t, enabled, "epoch manager not enabled")
+
+	countThreshold := mp.EpochParameters.EpochAdvanceOnCountThreshold
+	epochDuration := mp.EpochParameters.MinEpochDuration
+
+	err = env.Repository.Refresh(ctx)
+	require.NoError(t, err)
+
+	// write countThreshold index blobs: writing an object & flushing creates
+	// an index blob
+	for c := range countThreshold {
+		err = repo.WriteSession(ctx, env.Repository, repo.WriteSessionOptions{}, func(ctx context.Context, w repo.RepositoryWriter) (err error) {
+			ow := w.NewObjectWriter(ctx, object.WriterOptions{})
+			require.NotNil(t, ow)
+
+			defer func() {
+				cerr := ow.Close()
+				err = errors.Join(err, cerr)
+			}()
+
+			_, err = fmt.Fprintf(ow, "%v-%v", 0, c) // epoch count, object count
+			if err != nil {
+				return err
+			}
+
+			_, err = ow.Result() // force content write
+
+			return err
+		})
+
+		require.NoError(t, err)
+	}
+
+	// advance time and write more index to force epoch advancement on maintenance
+	ft.Advance(epochDuration + time.Second)
+	ow := env.RepositoryWriter.NewObjectWriter(ctx, object.WriterOptions{})
+	require.NotNil(t, ow)
+
+	_, err = fmt.Fprintf(ow, "%v-%v", 0, "last-object-in-epoch")
+	require.NoError(t, err)
+
+	_, err = ow.Result() // force content write
+	require.NoError(t, err)
+
+	err = ow.Close()
+	require.NoError(t, err)
+
+	// verify that there are enough index blobs to advance the epoch
+	epochSnap, err := emgr.Current(ctx)
+	require.NoError(t, err)
+
+	err = env.RepositoryWriter.Flush(ctx)
+	require.NoError(t, err)
+
+	require.Zero(t, epochSnap.WriteEpoch, "write epoch was advanced")
+	require.GreaterOrEqual(t, len(epochSnap.UncompactedEpochSets[0]), countThreshold, "not enough index blobs were written")
+
+	// verify quick maintenance has NOT run yet
+	sch, err := maintenance.GetSchedule(ctx, env.RepositoryWriter)
+
+	require.NoError(t, err)
+	require.True(t, sch.NextFullMaintenanceTime.IsZero(), "unexpected NextFullMaintenanceTime")
+	require.True(t, sch.NextQuickMaintenanceTime.IsZero(), "unexpected NextQuickMaintenanceTime")
+
+	err = snapshotmaintenance.Run(ctx, env.RepositoryWriter, maintenance.ModeQuick, false, maintenance.SafetyFull)
+	require.NoError(t, err)
```

Comment:
this could also probably be pulled out into a helper function if we wanted to minimize boiler plate

Reply:
Done
- #4191